cpp/models/ide_secir/parameters_io.cpp

```diff
@@ -0,0 +1,231 @@
+#include "memilio/utils/date.h"
+
+#include <string>
+#include <cmath>
+
+namespace mio
+{
+namespace isecir
+{
+
+// We assume that we start the simulation at time 0 and want to compute the necessary flows
+// in the past for the initialization of an IDE-SECIR model based on real data.
+IOResult<void> set_initial_flows(Model& model, ScalarType dt, std::string const& path, Date date,
+                                 ScalarType scale_confirmed_cases)
+{
+    //--- Preparations ---
+    // Try to get rki data from path.
+    BOOST_OUTCOME_TRY(rki_data, mio::read_confirmed_cases_noage(path));
+    auto max_date_entry = std::max_element(rki_data.begin(), rki_data.end(), [](auto&& a, auto&& b) {
+        return a.date < b.date;
+    });
+    if (max_date_entry == rki_data.end()) {
+        log_error("RKI data file is empty.");
+        return failure(StatusCode::InvalidFileFormat, path + ", file is empty.");
+    }
+    auto max_date = max_date_entry->date;
+    if (max_date < date) {
+        log_error("Specified date does not exist in RKI data.");
+        return failure(StatusCode::OutOfRange, path + ", specified date does not exist in RKI data.");
+    }
+
+    // Get (global) support_max to determine how many flows in the past we have to compute.
+    ScalarType global_support_max         = model.get_global_support_max(dt);
+    Eigen::Index global_support_max_index = Eigen::Index(std::ceil(global_support_max / dt));
+
+    // m_transitions should be empty at the beginning.
+    if (model.m_transitions.get_num_time_points() > 0) {
+        model.m_transitions = TimeSeries<ScalarType>((int)InfectionTransition::Count);
+    }
+
+    // The first time we need is -4 * global_support_max.
+    Eigen::Index start_shift = -4 * global_support_max_index;
+    // The last time needed is dependent on the mean stay times in the Exposed compartment and in InfectedNoSymptoms
+    // before transition to InfectedSymptoms.
+    Eigen::Index last_time_index_needed = Eigen::Index(std::ceil(
+        (model.parameters.get<TransitionDistributions>()[Eigen::Index(InfectionTransition::ExposedToInfectedNoSymptoms)]
+             .get_mean(dt) +
+         model.parameters
+             .get<TransitionDistributions>()[Eigen::Index(InfectionTransition::InfectedNoSymptomsToInfectedSymptoms)]
+             .get_mean(dt)) /
+        dt));
+    // Create TimeSeries with zeros.
+    for (Eigen::Index i = start_shift; i <= last_time_index_needed; i++) {
+        // Add time point.
+        model.m_transitions.add_time_point(
+            i * dt, TimeSeries<ScalarType>::Vector::Constant((int)InfectionTransition::Count, 0.));
+    }
+
+    //--- Calculate the flow InfectedNoSymptomsToInfectedSymptoms using the RKI data and store in the m_transitions object.---
+    ScalarType min_offset_needed = std::ceil(
+        model.m_transitions.get_time(0) -
+        1); //Need -1 if first time point is integer and just the floor value if not, therefore use ceil and -1
```

Comment:
```suggestion
        1); //Need -1 if first time point is integer and just the ceil value if not, therefore use ceil and -1
```
Is this what you mean?

Reply:
Actually, floor is right here. If model.m_transitions.get_time(0) is not an integer, i need floor(model.m_transitions.get_time(0)) for the calculation which is equal to ceil(model.m_transitions.get_time(0)-1).
If model.m_transitions.get_time(0) is an integer, i need model.m_transitions.get_time(0)-1 for the calculation.